### DISCUSSION TITLE
Yarn install fails in github Action

### DISCUSSION THREAD

**Discussion by yamipunetha** [community/community#132650] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi everyone, 

I have a react project, which is deployed via Github action. 

The yarn install step failed suddenly today, we did not change anything in package.json. 

name: Install the dependencies
run: |
  yarn install --frozen-lockfile
  
![Screenshot 2024-07-18 at 15 20 10](https://github.com/user-attachments/assets/0bbec223-d5f0-49bb-8d71-9b3330322910)

It is working locally, just not in Github Actions.

Thank you in advance.

**Comment by lucaong** [comment]
Same issue here. It looks like the action proceeds without waiting for `yarn install --frozen-lockfile` to complete or fail. Consequently, later steps fail because of missing dependencies.

An example: https://github.com/lucaong/minisearch/actions/runs/9992214982/job/27616688415

<img width="1261" alt="Screenshot 2024-07-18 at 4 32 07 PM" src="https://github.com/user-attachments/assets/9b53564e-f711-4e2c-adb8-559610a237e2">

It seems to occur only with the latest NodeJS `v22.5.0`, released yesterday, therefore it affects projects using `actions/setup-node` and setting `node-version: latest`. Setting `node-version: 21` seems to fix it.

**Reply by lucaong** [reply]
This is likely related to https://github.com/yarnpkg/berry/issues/6398, which in turn is probably due to https://github.com/nodejs/node/issues/53902

**Comment by lucaong** [comment, marked as answer]
As mentioned in my comment above, this is likely related to using `actions/setup-node` with `node-version: latest` and therefore hitting [a bug](https://github.com/nodejs/node/issues/53902) in NodeJS `v22.5.0` (released yesterday).

Solution:
  - For the moment, set `node-version: 21` in your workflow
  - As soon as NodeJS releases a new patched version, it is possible to revert to `node-version: latest`

**Reply by yamipunetha** [reply]
Thank you! :D

**Reply by jovana** [reply]
Many thanks!!!
